File: python-project/experiments/mixed/RunMixedExperiments.py
from experiments.mixed import Exp_autos
from experiments.mixed import Exp_census_india
from experiments.mixed import Exp_ecoli
from experiments.mixed import Exp_haberman
from experiments.mixed import Exp_housing_prices
from experiments.mixed import Exp_ionosphere
from experiments.mixed import Exp_iris
from experiments.mixed import Exp_parkinsons
from experiments.mixed import Exp_planning_relax
from experiments.mixed import Exp_qsar_biodeg
from experiments.mixed import Exp_thoracic_surgery
from experiments.mixed import Exp_thyroid
from experiments.mixed import Exp_user_knowledge
from experiments.mixed import Exp_vehicle
from experiments.mixed import Exp_vertebral


def main():
    run = 1
    n_folds = 10
    fold_log = True
    kde_bw = "normal_reference"

    exp = Exp_haberman.Exp_haberman("haberman")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_iris.Exp_iris("iris")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_user_knowledge.Exp_user_knowledge("user_knowledge")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_vertebral.Exp_vertebral("vertebral")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_ecoli.Exp_ecoli("ecoli")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_planning_relax.Exp_planning_relax("planning_relax")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_thoracic_surgery.Exp_thoracic_surgery("thoracic_surgery")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_vehicle.Exp_vehicle("vehicle")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_thyroid.Exp_thyroid("thyroid")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_parkinsons.Exp_parkinsons("parkinsons")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_autos.Exp_autos("autos")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_ionosphere.Exp_ionosphere("ionosphere")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_qsar_biodeg.Exp_qsar_biodeg("qsar_biodeg")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_housing_prices.Exp_housing_prices("housing_prices")
    exp.run(run, n_folds, fold_log, kde_bw)

    exp = Exp_census_india.Exp_census_india("census_india")
    exp.run(run, n_folds, fold_log, kde_bw)


if __name__ == "__main__":
    main()

File: src/main/java/experiments/missing/mixed/Run_mixed_experiments.java
package experiments.missing.mixed;

import eu.amidst.extension.learn.parameter.InitializationTypeVBEM;
import eu.amidst.extension.learn.parameter.InitializationVBEM;
import eu.amidst.extension.learn.parameter.VBEMConfig;
import eu.amidst.extension.learn.parameter.penalizer.NoPenalizer;
import eu.amidst.extension.learn.structure.hillclimber.BayesianHcConfig;
import eu.amidst.extension.util.LogUtils;
import methods.missing.MeanImputation;
import methods.missing.MissingMixedMethod;

import java.util.LinkedHashSet;
import java.util.Set;

public class Run_mixed_experiments {

    public static void main(String[] args) throws Exception {
        long seed = 0;
        int nRuns = 5;
        boolean storeModels = false;

        LogUtils.LogLevel logLevel = LogUtils.LogLevel.INFO;
        int maxNumberOfParents = 3;
        Set<MissingMixedMethod> methods = new LinkedHashSet<>();

        /* Missing VBSEM configuration */
        InitializationVBEM initializationVBEM = new InitializationVBEM(InitializationTypeVBEM.PYRAMID, 64, 16, false);
        VBEMConfig vbemConfig = new VBEMConfig(seed, 0.01, 100, initializationVBEM, new NoPenalizer());
        BayesianHcConfig bayesianHcConfig = new BayesianHcConfig(seed, 0.01, 100);

        /* Methods */
        //methods.add(new MissingVBSEM(vbemConfig, bayesianHcConfig, 100, maxNumberOfParents));
        methods.add(new MeanImputation());

        /* Run experiments */
        for(double missingPercentage = 0.1; missingPercentage < 0.6; missingPercentage += 0.1) {

            Exp_haberman exp_haberman = new Exp_haberman(methods);
            exp_haberman.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_iris exp_iris = new Exp_iris(methods);
            exp_iris.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_user_knowledge exp_user_knowledge = new Exp_user_knowledge(methods);
            exp_user_knowledge.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_vertebral exp_vertebral = new Exp_vertebral(methods);
            exp_vertebral.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_ecoli exp_ecoli = new Exp_ecoli(methods);
            exp_ecoli.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_planning_relax exp_planning_relax = new Exp_planning_relax(methods);
            exp_planning_relax.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_thoracic_surgery exp_thoracic_surgery = new Exp_thoracic_surgery(methods);
            exp_thoracic_surgery.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_vehicle exp_vehicle = new Exp_vehicle(methods);
            exp_vehicle.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_thyroid exp_thyroid = new Exp_thyroid(methods);
            exp_thyroid.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_parkinsons exp_parkinsons = new Exp_parkinsons(methods);
            exp_parkinsons.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_autos exp_autos = new Exp_autos(methods);
            exp_autos.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_ionosphere exp_ionosphere = new Exp_ionosphere(methods);
            exp_ionosphere.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_qsar_biodeg exp_qsar_biodeg = new Exp_qsar_biodeg(methods);
            exp_qsar_biodeg.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_housing_prices exp_housing_prices = new Exp_housing_prices(methods);
            exp_housing_prices.runExperiment(missingPercentage, nRuns, storeModels, logLevel);

            Exp_census_india exp_census_india = new Exp_census_india(methods);
            exp_census_india.runExperiment(missingPercentage, nRuns, storeModels, logLevel);
        }
    }
}
